tests/grid/cell_id_06.cc

```diff
@@ -0,0 +1,49 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// testing parent and child relationship of CellIds
+
+#include <deal.II/grid/cell_id.h>
+
+#include "../tests.h"
+
+int
+main()
+{
+  initlog();
+
+  // create some arbitrary CellIds
+  CellId id0(0, {0});
+  CellId id1(0, {1});
+  CellId id2(0, {0, 0});
+  CellId id3(0, {0, 0, 0});
+  CellId id4(1, {0});
+
+  // same cell
+  AssertThrow(id0.is_parent_of(id0) == false, ExcInternalError());
+
+  // same level
+  AssertThrow(id0.is_parent_of(id1) == false, ExcInternalError());
+
+  // child
+  AssertThrow(id0.is_parent_of(id2) == true, ExcInternalError());
+
+  // grand child
+  AssertThrow(id0.is_parent_of(id3) == false, ExcInternalError());
+
+  // cell with different coarse-cell id
+  AssertThrow(id0.is_parent_of(id4) == false, ExcInternalError());
```

Comment:
Do you think it would make sense to output the relations `id0.is_parent_of(...)` to the logfile rather than assert it? At least to me errors would be easier to read than by having those conditions hardcoded. But I'll let @bangerth express an opinion as well.

Reply:
Yes, I agree. Debugging failing `AssertThrow` conditions is awkward. Just do
```
  deallog << std::boolalpha;
  deallog << id0.is_parent_of(id0) << std::endl;
  deallog << id0.is_parent_of(id1) << std::endl;
  ...
```